### PR TITLE
feat: add redirect capability migration

### PR DESCRIPTION
**Pull request by foodaka**
## General Changes
Redirects users to the v3 market dashboard after migration

## Developer Notes

Add any notes here that may be helpful for reviewers.

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Comment by foodaka** [comment]
Link T-5768

**Review by defispartan — APPROVED** [review]
V3DashboardButton

Looks good and works great!

**Comment by MareskoY** [comment]
**Bug1:** 
redirection every time to mainnet _/?marketName=proto_mainnet_v3_
if we migrate from a polygon or from an avalanche we any way redirecting to mainnet

**Bug2:** 
after redirection, we clean up local storage and disconnect from the wallet also if it's a fork we also delete the fork capability

**Comment by MareskoY** [comment]
> **Bug1:** redirection every time to mainnet _/?marketName=proto_mainnet_v3_ if we migrate from a polygon or from an avalanche we any way redirecting to mainnet
> 
> **Bug2:** after redirection, we clean up local storage and disconnect from the wallet also if it's a fork we also delete the fork capability

this issue only with cypress in usual browsers all is ok so pased

**Review by defispartan — APPROVED** [review]

File: src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx [hunk 1]
```diff
@@ -68,7 +70,34 @@ export const MigrateV3ModalContent = () => {
   if (txError && txError.blocking) {
     return <TxErrorView txError={txError} />;
   }
-  if (migrateTxState.success) return <TxSuccessView action={<Trans>Migrated</Trans>} />;
+
+  const handleRoute = () => {
+    if (currentMarket === 'proto_polygon') {
+      setCurrentMarket('proto_polygon_v3' as CustomMarket);
+      window.location.href = '/?marketName=proto_polygon_v3';
```
Comment (grothem) [inline src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx:77]: You could use the CustomMarket enum values directly here, might be a little cleaner. Something like:
```ts
if (currentMarket === CustomMarket.proto_polygon) {
       setCurrentMarket(CustomMarket.proto_polygon_v3);
       window.location.href = `/?marketName=${CustomMarket.proto_polygon_v3}`;
} else if ...
```

**Review by grothem — COMMENTED** [review]
Looks good, just one suggestion, and also need to merge latest in from main to fix merge conflicts

File: src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx (hunk 1 above)
Comment (foodaka) [reply src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx]: good catch! added here `ed5bf838ccfb929a92fbc44baa5602c76e4ea670`

File: src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx [hunk 2]
```diff
@@ -68,7 +70,34 @@ export const MigrateV3ModalContent = () => {
   if (txError && txError.blocking) {
     return <TxErrorView txError={txError} />;
   }
-  if (migrateTxState.success) return <TxSuccessView action={<Trans>Migrated</Trans>} />;
+
+  const handleRoute = () => {
+    if (currentMarket === CustomMarket.proto_polygon) {
```
Comment (foodaka) [inline src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx:75]: @MareskoY the fork tests didnt work on polygon / avax cause currrent market says `polygon_fork` --> i dont think it makes sense to handle this clause wdyt? cc @grothem ?

Comment (foodaka) [reply src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx:75]: do we usually test against this case?

Screenshot 2023-02-13 at 19 24 18

Comment (MareskoY) [reply src/components/transactions/MigrateV3/MigrateV3ModalContent.tsx:75]: > @MareskoY the fork tests didnt work on polygon / avax cause currrent market says `polygon_fork` --> i dont think it makes sense to handle this clause wdyt? cc @grothem ?

it's not necessary functionality for testing (e2e uses another way to switch markets)

but if it is easy to fix for the forks will be good to fix

**Review by grothem — APPROVED** [review]

**Review by MareskoY — APPROVED** [review]

**Pull request merged**